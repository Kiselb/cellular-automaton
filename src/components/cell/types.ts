export type Params = {
  generation: number;
  row: number;
  col: number;
  colorGammaMin?: string;
  colorGammaMax?: string;
  colorEmpty: string;
  colorCell: string;
};

export type Props = Omit<Params, "colorCell"> & {
  onClick: (cell: Params) => void;
  probe?: ({}) => void;
};

export type CalcColor = (
  generation: number,
  colorScaleMin?: string,
  colorScaleMax?: string
) => string;

export { Cell, calcCellColor } from "./Cell";
