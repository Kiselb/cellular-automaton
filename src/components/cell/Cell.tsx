import React, { FC } from "react";
import chroma from "chroma-js";

import { Params, Props, CalcColor } from "./types";

//import "./cell.css";
import styles from "./cell.module.css";

export const calcCellColor: CalcColor = (
  generation: number,
  colorScaleMin = "yellow",
  colorScaleMax = "darkgreen"
) =>
  chroma
    .scale([colorScaleMin, colorScaleMax])((generation - 1) * 0.1)
    .hex();

export const Cell: FC<Props> = ({
  generation,
  row,
  col,
  onClick,
  colorEmpty,
  probe,
}: Props) => {
  const colorCell: string = calcCellColor(generation);
  const cellParams: Params = {
    generation,
    row,
    col,
    colorEmpty,
    colorCell,
  };
  const onClickHandler = (event: React.MouseEvent<HTMLDivElement>) => {
    event.preventDefault();
    onClick(cellParams);
  };
  const onMouseOverHandler = (event: React.MouseEvent<HTMLDivElement>) => {
    event.preventDefault();
    event.buttons === 1 && onClick(cellParams);
    event.buttons === 1 && probe && probe(cellParams);
  };
  const onDoubleClickHandler = (event: React.MouseEvent<HTMLDivElement>) => {
    event.preventDefault();
    probe && probe(cellParams);
  };
  return (
    <div
      key={`R${row}:C${col}`}
      data-testid={`R${row}:C${col}`}
      className={styles["cell"]}
      onClick={onClickHandler}
      onDoubleClick={onDoubleClickHandler}
      onMouseOver={onMouseOverHandler}
      style={{ background: generation === 0 ? colorEmpty : colorCell }}
    ></div>
  );
};
