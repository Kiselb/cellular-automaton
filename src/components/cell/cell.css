.cell {
    height: 10px;
    width: 10px;
    margin: 0;
}
.cell--dead {
    background-color: #ececec;
}
.cell--alive {
    background-color: green;
}
.cell--old {
    background-color: lightgreen;
}
