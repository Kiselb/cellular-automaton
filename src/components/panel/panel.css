.panel {
    display: inline-block;
    background-color: lightslategrey;
    padding-left: 5px;
    padding-right: 6px;
    padding-top: 5px;
    padding-bottom: 6px;
    border: 1px solid black;

}
.grid--row {
    display: flex;
    width: 100%;
    white-space: nowrap;
    padding: 0;
    margin: 0;
    font-size: 0;
}
.grid--cell {
    display: inline-block;
    height: 10px;
    width: 10px;
    border-top: 1px solid lightslategrey;
	border-left: 1px solid lightslategrey;    
    padding: 0;
    margin: 0;
}
