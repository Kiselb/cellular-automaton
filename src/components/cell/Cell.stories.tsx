import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Cell } from "./Cell";

export default {
  title: "Application/Cell",
  component: Cell,
  argTypes: {
    colorDead: { control: "color" },
    colorAlive: { control: "color" },
    colorOld: { control: "color" },
  },
} as ComponentMeta<typeof Cell>;

const Template: ComponentStory<typeof Cell> = (args) => <Cell {...args} />;

export const Dead = Template.bind({});
Dead.args = {
  status: "dead",
  row: 13,
  col: 13,
  onClick: () => console.log("OK"),
  colorDead: "#ececec",
  colorAlive: "lightgreen",
  colorOld: "green",
};

export const Alive = Template.bind({});
Alive.args = {
  status: "alive",
  row: 13,
  col: 13,
};

export const Old = Template.bind({});
Old.args = {
  status: "old",
  row: 13,
  col: 13,
};

export const OldClickable = Template.bind({});
OldClickable.args = {
  status: "old",
  row: 13,
  col: 13,
  onClick: () => alert("R13C13"),
};

export const Colored = Template.bind({});
Colored.args = {
  status: "old",
  row: 13,
  col: 13,
  onClick: () => alert("R13C13"),
  colorDead: "blue",
  colorAlive: "red",
  colorOld: "green",
};
