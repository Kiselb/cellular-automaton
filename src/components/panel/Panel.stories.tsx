import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Panel } from "./types";
import { Cell } from "../cell/types";

export default {
  title: "Application/Panel",
  component: Panel,
  argTypes: {},
  subcomponents: { Cell },
} as ComponentMeta<typeof Panel>;

const Template: ComponentStory<typeof Panel> = (args) => <Panel {...args} />;

const mock10: number[][] = [];
for (let i = 0; i < 10; i++) {
  const row: number[] = [];
  for (let j = 0; j < 10; j++) {
    row[j] = j;
  }
  mock10[i] = row;
}

const mock40: number[][] = [];
for (let i = 0; i < 40; i++) {
  const row: number[] = [];
  for (let j = 0; j < 40; j++) {
    row[j] = j;
  }
  mock40[i] = row;
}

const mock75: number[][] = [];
for (let i = 0; i < 75; i++) {
  const row: number[] = [];
  for (let j = 0; j < 75; j++) {
    row[j] = j;
  }
  mock75[i] = row;
}

export const Size10x10 = Template.bind({});
Size10x10.args = {
  data: mock10,
};

export const Size40x40 = Template.bind({});
Size40x40.args = {
  data: mock40,
};

export const Size75x75 = Template.bind({});
Size75x75.args = {
  data: mock75,
};
