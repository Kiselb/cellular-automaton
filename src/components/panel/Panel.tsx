import React from "react";
import "./panel.css";
import { Cell, CellParams } from "../cell/Cell";

interface PanelProps {
  data: number[][];
  onChange: (params: CellParams) => void;
}

export const Panel = ({ data, onChange, ...props }: PanelProps) => {
  const cellOnClick = (params: CellParams) => {
    onChange(params);
  };
  return (
    <div className={["panel"].join(" ")} {...props}>
      {data.map((row, rowindex) =>
        row.map((item, colindex) => (
          <div
            key={`Wrap:R${rowindex}:C${colindex}`}
            className={[
              "grid",
              "grid--cell",
              colindex % row.length === 0 ? "grid--wrap" : "",
            ].join(" ")}
            data-testid="Panel"
          >
            <Cell
              data-testid={`R${rowindex}:C${colindex}`}
              row={rowindex}
              col={colindex}
              generation={item}
              onClick={cellOnClick}
              colorEmpty="#ececec"
              colorGamma="#77FF77"
            ></Cell>
          </div>
        ))
      )}
    </div>
  );
};
