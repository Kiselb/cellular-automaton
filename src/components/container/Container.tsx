import React, { useEffect, useState } from "react";
import "./container.css";
import { CellParams, CellStatus } from "../cell/Cell";
import { Panel, PanelMode } from "../panel/Panel";

interface ContainerProps {
  rows: number;
  cols: number;
  probe?: (params: any) => void;
}

export const Container = ({ rows, cols, ...props }: ContainerProps) => {
  const [data, setData] = useState<CellStatus[][]>([]);
  const mode: PanelMode = "paused";
  const onChangeHandler = (params: CellParams) => {
    data[params.row][params.col] = params.status === "dead" ? "alive" : "dead";
    setData([...data]);
    props.probe && props.probe({ rows, cols });
  };
  useEffect(() => {
    const blank: CellStatus[][] = Array.from({ length: rows }, () =>
      Array.from({ length: cols }, () => "dead")
    );
    setData(blank);
  }, [rows, cols]);
  return (
    <div className={["container"].join(" ")} {...props}>
      <Panel data={data} mode={mode} onChange={onChangeHandler} {...props} />
    </div>
  );
};
