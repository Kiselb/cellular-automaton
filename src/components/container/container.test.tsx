import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import { CellStatus } from "../cell/Cell";
import { Panel, PanelMode } from "../panel/Panel";
import { Container } from "./Container";

it("Container renders correctly", () => {
  const { asFragment } = render(<Container rows={10} cols={10} />);

  screen.debug();
  expect(asFragment()).toMatchSnapshot();
});
it("Container has cell R1:C1", () => {
  render(<Container rows={10} cols={10} />);

  const div = screen.getByTestId("R1:C1") as HTMLDivElement;
  expect(div).toBeTruthy();
});
it("Container update rows and cols", () => {
  render(<Container rows={100} cols={100} />);

  const div = screen.getByTestId("R99:C99") as HTMLDivElement;
  expect(div).toBeTruthy();
});
it("Container event handler", () => {
  const probe = jest.fn();
  render(<Container rows={100} cols={100} probe={probe} />);

  const div = screen.getByTestId("R1:C1") as HTMLDivElement;
  fireEvent.click(div);
  fireEvent.click(div);
  fireEvent.click(div);
  fireEvent.click(div);
  fireEvent.click(div);
  fireEvent.click(div);
  fireEvent.click(div);
  fireEvent.click(div);
  fireEvent.click(div);
  fireEvent.click(div);
  expect(probe).toBeCalledTimes(10);
});
