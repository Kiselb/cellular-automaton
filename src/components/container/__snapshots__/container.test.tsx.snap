// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="panel"
    >
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R0:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R1:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R2:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R3:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R4:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R5:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R6:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R7:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R8:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
      <div
        class="grid--row"
      >
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C0"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C1"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C2"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C3"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C4"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C5"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C6"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C7"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C8"
            style="background: rgb(236, 236, 236);"
          />
        </div>
        <div
          class="grid grid--cell"
          data-testid="Panel"
        >
          <div
            class="cell"
            data-testid="R9:C9"
            style="background: rgb(236, 236, 236);"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
